### DISCUSSION TITLE
Help regarding nwbwidget( IVCurveWidget and SingleTracePlotlyWidget ) .

### DISCUSSION THREAD

**Discussion by Daksh1603** [NeurodataWithoutBorders/helpdesk#50] [Q&A] [answered]
Hello,

I am currently a GSoC intern working on developing a python notebook that helps in exploring an ephys NWB file and I am vastly making using of the NWB Widgets as they are really cool and almost cover everything I plan to achieve. I just have few basic doubts and thought this platform is the right place to post. If there is any other alternative platform where I can ask for help , please do let me know.

1) **SingleTracePlotlyWidget** : In order to visualize traces of both the acquisition and stimulus series, I am making use of this widget by creating it with the first trace and just adding the rest using the add_trace function. I feel there might be a better alternative that I have overlooked but I did try experimenting with SeparateTracesPlotlyWidget and AlignMultiTraceTimeSeriesByTrialsConstant ( both from timeseries.py ) but they didn't seem to work eventually. The latter widget couldn't find a trials section in the file and was running into "No trials present" . The approach that I have implemented works for multiple traces but it becomes rather slow when the user wants to select a few particular traces for display. 

2) **IVCurveWidget** : I plan on adding this widget to the notebook, but I am not able to figure out as to how to get this work . Perhaps I should further explore a little more into DynamicTableRegions for a nwbfile to solve both these issues but a little direction would help a lot.

I am still a beginner trying to understand more about this, the nwb file format in general but would love to learn, help and contribute more! Thanking you in anticipation

**Comment by bendichter** [comment]
@luiztauffer

**Comment by sanjayankur31** [comment]
Thanks @Daksh1603 . Just CC'ing @pgleeson and myself (mentors)

**Comment by luiztauffer** [comment, marked as answer]
Hi @Daksh1603 @sanjayankur31
great to see you’re using the widgets! Here are some ideas that might help you out:

- try running the entire nwb file using either `nwb2widget` or `Panel`, [reference](https://nwb-widgets.readthedocs.io/en/latest/contents/running_on_jupyter.html). This will automatically render the correct widget for each data type within that file, this can serve as an initial test, to check if the file is formatted as the widgets expect
- try using `timeseries.show_timeseries()` or `timeseries.BaseGroupedTraceWidget()` for multiple traces
- The `IVCurveWidget` expects a `SequentialRecordingsTable` object as input, is that what you’re passing to the widget? [Here’s a tutorial](https://pynwb.readthedocs.io/en/stable/tutorials/domain/plot_icephys.html#sphx-glr-tutorials-domain-plot-icephys-py) on how to organize icephys data on nwb

Feel free to share any bugs you might find and further questions directly on nwgwidgets repository: https://github.com/NeurodataWithoutBorders/nwbwidgets

**Comment by Daksh1603** [comment]
Hello @luiztauffer,
Thanks a lot for the help ! I will certainly look into all the points you mentioned. As I explore more of nwbwidgets, Ill try my best to make meaningful contributions as well : )
